### PR TITLE
Add a run history to tasks and notify input-files reception on new rabbitmq binding

### PR DESCRIPTION
**Pull request by vbochetRTE**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: gridcapa-task-manager-api/src/test/java/com/farao_community/farao/gridcapa/task_manager/api/ProcessFileDtoTest.java [hunk 1]
```diff
@@ -25,4 +30,32 @@ void testConstructorEmptyProcessFile() {
...
+        OffsetDateTime now = OffsetDateTime.now();
+        ProcessFileDto referenceDto = new ProcessFileDto("path/to/filename", "FILETYPE", ProcessFileStatus.VALIDATED, "filename", now);
+        ProcessFileDto equalDto = new ProcessFileDto("path/to/filename", "FILETYPE", ProcessFileStatus.VALIDATED, "filename", now);
+        ProcessFileDto diffFilePathDto = new ProcessFileDto("other/path/to/filename", "FILETYPE", ProcessFileStatus.VALIDATED, "filename", now);
+        ProcessFileDto diffFileTypeDto = new ProcessFileDto("path/to/filename", "DIFFERENT", ProcessFileStatus.VALIDATED, "filename", now);
+        ProcessFileDto diffStatusDto = new ProcessFileDto("path/to/filename", "FILETYPE", ProcessFileStatus.NOT_PRESENT, "filename", now);
+        ProcessFileDto diffFilenameDto = new ProcessFileDto("path/to/filename", "FILETYPE", ProcessFileStatus.VALIDATED, "other", now);
+        ProcessFileDto diffDateDto = new ProcessFileDto("path/to/filename", "FILETYPE", ProcessFileStatus.VALIDATED, "filename", now.minusMinutes(1));
+
+        assertFalse(referenceDto.equals(null)); // do not use assertFalse, which does not call equals method in this case
```
Comment (kahyami) [inline gridcapa-task-manager-api/src/test/java/com/farao_community/farao/gridcapa/task_manager/api/ProcessFileDtoTest.java:45]: in the comment you put do not use "assertFalse...."   and you actually used it !

File: gridcapa-task-manager-api/src/test/java/com/farao_community/farao/gridcapa/task_manager/api/ProcessFileDtoTest.java [hunk 2]
```diff
@@ -25,4 +30,32 @@ void testConstructorEmptyProcessFile() {
...
+        assertNotEquals(referenceDto, new Object()); // do not use assertFalse, which does not call equals method in this case
+        assertTrue(referenceDto.equals(referenceDto)); // do not use assertFalse, which does not call equals method in this case
```
Comment (kahyami) [inline gridcapa-task-manager-api/src/test/java/com/farao_community/farao/gridcapa/task_manager/api/ProcessFileDtoTest.java:47]: the same here

File: gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/TaskService.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+    public Task addNewRunAndSaveTask(OffsetDateTime timestamp) {
+        Task task = taskRepository.findByTimestamp(timestamp).orElseThrow(TaskNotFoundException::new);
+        ProcessRun processRun = new ProcessRun(task.getProcessFiles().stream().filter(ProcessFile::isInputFile).toList());
+        task.addProcessRun(processRun);
+        return taskRepository.save(task);
+    }
+
+    static void removeUnavailableProcessFileFromTaskRunHistory(ProcessFile processFile, Task task, FileEventType fileEventType) {
+        if (FileEventType.UPDATED.equals(fileEventType) || FileEventType.DELETED.equals(fileEventType)) {
```
Comment (danielthirion) [inline gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/TaskService.java:232]: Perhaps use "==" to compare Enum, as you've done L213 & L217, so it is coherent across the class (or change it the other way around)

**Review by danielthirion — DISMISSED** [review]

File: gridcapa-task-manager-app/src/test/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManagerControllerTest.java [hunk 4]
```diff
@@ -138,6 +143,25 @@ void testUpdateOk() {
...
+        Task task = new Task(taskTimestamp);
+        Mockito.when(taskService.addNewRunAndSaveTask(Mockito.any())).thenReturn(task);
+        ResponseEntity<TaskDto> response = taskManagerController.addNewRunInTaskHistory(taskTimestamp.toString());
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertNotNull(response.getBody());
+        assertEquals(taskTimestamp, response.getBody().getTimestamp());
+    }
+
+    @Test
+    void testSAddNewRunInTaskHistoryThrowsTaskNotFoundException() {
```
Comment (kahyami) [inline gridcapa-task-manager-app/src/test/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManagerControllerTest.java:158]: Fix the name of the method

**Review by danielthirion — APPROVED** [review]

**Review by danielthirion — APPROVED** [review]

**Review by danielthirion — DISMISSED** [review]

**Pull request merged**